tests/test_trainer.py

```diff
@@ -34,6 +34,11 @@ def output_logs_dir():
         yield temp_dir
 
 
+@pytest.fixture('session')
+def catdog_num_classes():
```

Comment:
Small nitpick: can you remove this fixture and just use `len(catdog_dictionary)` in the assignment?